code/modules/mining/machine_processing.dm

```diff
@@ -215,98 +229,157 @@
 	var/selected_iron = 0
 	var/selected_clown = 0
 	var/selected_titanium = 0
+	var/selected_bluespace = 0
 	var/on = 0 //0 = off, 1 =... oh you know!
 
+/obj/machinery/mineral/processing_unit/Initialize()
+	. = ..()
+	proximity_monitor = new(src, 1)
+
+/obj/machinery/mineral/processing_unit/HasProximity(atom/movable/AM)
+	if(istype(AM, /obj/item/weapon/ore) && AM.loc == get_step(src, input_dir))
+		process_ore(AM)
+
+/obj/machinery/mineral/processing_unit/proc/process_ore(O)

```

Comment:
Oh no, I forgot to type it, fixing this asap.
Or maybe move this proc, along with the stacker's, up a level to the parent.